test/CoreTest.cpp

```diff
@@ -71,6 +71,56 @@ TEST_CASE( "attribute_dtype_test", "[core]" )
     REQUIRE(Datatype::ARR_DBL_7 == a.dtype);
     a = Attribute(static_cast< bool >(false));
     REQUIRE(Datatype::BOOL == a.dtype);
+
+    // translation from non-fixed size integer types
+    a = Attribute(static_cast< short >(1));
+    REQUIRE((
+        Datatype::INT16 == a.dtype || Datatype::INT32 == a.dtype || Datatype::INT64 == a.dtype
```

Comment:
[```determineDatatype()```](https://github.com/openPMD/openPMD-api/blob/0.4.0-alpha/include/openPMD/Datatype.hpp#L103-L140) provides the datatype of explicit fixed width you are looking for. Would be a little easier to read and more explicit.
```cpp
a = Attribute(static_cast< short >(1));
REQUIRE(determineDatatype< short >() == a.dtype);
```

Reply:
Sounds legit, but wouldn't we then test `determineDatatype()` only against itself?

Anyway, the real test this adds is about the line above, if it compiles: `a = Attribute(static_cast< short >(1));`

Reply:
> wouldn't we then test determineDatatype() only against itself?

No, because `Attribute.datatype` is *not* determined by `determineDatatype()`, but by (`std::`/`mpark::`)[`Variant`](https://github.com/openPMD/openPMD-api/blob/0.4.0-alpha/include/openPMD/auxiliary/Variant.hpp#L58) with a subsequent lookup in the `Datatype` enum.